File: sorting/insertion.cpp
#include<iostream>
using namespace std;

int insertion(int arr[],int n)
{
    for(int i=0;i<n;i++)
    {
        int key,j;
        arr[i]=key;
        j=i-1;
        while(j>=0 && arr[j]>key)
        {
            arr[j+1]=arr[j];
            j-=1;
        }
        arr[j+1]=key;

    }
    return *arr;
}


int main()
{
    int arr[]={1,3,2,5,4};
    int n=sizeof(arr);
    cout<<insertion(arr,n);
}

File: sorting/insertion.py
#----------------Insertion sort in python
import time
start_time=time.time()
def insertion(arr,n):
    for i in range(n):
        key=arr[i]
        j=i-1
        while (j>=0 and arr[j]>key):
            arr[j+1]=arr[j]
            j-=1
        arr[j+1]=key
    return arr

print(insertion([2,1,3,5,4],5))   
print("--- %s seconds ---" % (time.time() - start_time))                     


